h/views/admin_groups.py

```diff
@@ -45,18 +45,36 @@ def get(self):
     @view_config(request_method='POST')
     def post(self):
         def on_success(appstruct):
-            read_url = self.request.route_url('admin_groups')
-            self.request.session.flash('TODO: I will add a {gtype} group called "{name}"'
-                                       ' for authority {authority}, created by {creator}'
-                                       ' and origins {origins}'.format(
-                                            gtype=appstruct['group_type'],
-                                            name=appstruct['name'],
-                                            authority=appstruct['authority'],
-                                            creator=appstruct['creator'],
-                                            origins=', '.join(appstruct['origins'])
-                                       ), queue='success')
-            response = HTTPFound(location=read_url)
-            return response
+            svc = self.request.find_service(name='group')
+
+            # Create the new group.
+            creator = appstruct['creator']
+            authority = appstruct['authority']
+            description = appstruct['description']
+            name = appstruct['name']
+            origins = appstruct['origins']
+            type_ = appstruct['group_type']
+
+            userid = 'acct:{}@{}'.format(creator, authority)
+
+            if type_ == 'open':
+                group = svc.create_open_group(name=name, userid=userid,
+                                              origins=origins, description=description)
+            elif type_ == 'restricted':
+                group = svc.create_restricted_group(name=name, userid=userid,
+                                                    origins=origins, description=description)
```

Comment:
That seems like a bug in the `create_*` service methods? AIUI unscoped groups should fundamentally be allowed in our data model, even if they won't actually show up anywhere at the moment and thus aren't really useful.

Reply:
I dunno, is it a bug in the `create_*()` methods? They take a non-optional `origins` argument that's documented as being a list. If you pass in `None` (or some other non-list) they crash.

I _think_ they would work if you passed in an empty list rather than `None`, and you'd get an "unscoped" open or restricted group, but what I think unscoped would mean in practice is that the group never appears in the API / client on any site. I think this ability to pass in an empty list is more accident than design though.

I think for simplicity we should not allow open or restricted groups without scopes to be created yet, we don't know how such groups would behave or how we'd want them to behave. At the model / db level it's capable of representing them but at all other levels (service and above) their behaviour is undefined.